core/arch/proc_api.h

```diff
@@ -255,6 +265,48 @@ typedef enum {
     FEATURE_AVX512F = 16 + 128,  /**< AVX-512F instructions supported */
     FEATURE_AVX512BW = 30 + 128, /**< AVX-512BW instructions supported */
 } feature_bit_t;
+#elif defined(AARCHXX)
+/* On Arm, architectural features are defined and stored very differently from
+ * X86. Specifically:
+ * - There are multiple 64 bit system registers for features storage only, FREG.
+ * - Each register is divided into nibbles representing a feature, NIBPOS.
+ * - The value of a nibble represents a certain level of support for that feature, FVAL.
+ * - The values can range from 0 to 15. In most cases 0 means a feature is not
+ *   supported at all but in some cases 15 means a feature is not supported at
+ *   all, NSFLAG.
+ */
+#    define DEF_FEAT(FREG, NIBPOS, FVAL, NSFLAG) \
+        ((ushort)((NSFLAG << 15) | (FREG << 8) | (NIBPOS << 4) | FVAL))
+#    define GET_FEAT_REG(FEATURE) (feature_reg_idx_t)((((ushort)FEATURE) & 0x7F00) >> 8)
```

Comment:
Normally the _api.h files only contain public, doxygen-documented items.  This and the next few should probably be in proc.h or just proc.c instead.

Reply:
Moved to proc.c.